src/lib/services/workflow-service.ts

```diff
@@ -121,9 +124,24 @@ export async function fetchWorkflowWithEventHistory(
   parameters: GetWorkflowExecutionRequest,
   request = fetch,
 ): Promise<{ workflow: WorkflowExecution; events: HistoryEventWithId[] }> {
+  let port = get(webDecoderPort);
+  // if port rawPayloads = true
+
   const [workflow, events] = await Promise.all([
     fetchWorkflow(parameters, request),
-    fetchEvents(parameters, request).then(toEventHistory),
+    fetchEvents(parameters, request)
+      .then(toEventHistory)
+      .then((events) => {
```

Comment:
We seem to return `events` no matter what?

Reply:
If you make this an `async`/`await`, then I think you can spare yourself all of the `Promise.resolve`s. Not a blocker. This code could probably use some refactoring, but it's not our biggest priority at this moment.

Reply:
Yeah the convertEventPayloadFromDataConverter mutates the events object (not ideal) but the way the code was structured made this less than ideal. We can update this with a better access pattern using some more of Svelte's primitives in the future. I'm 100% in agreement this is not great code.

We've talked about what we want the future state to be and it doesn't really look anything like this.